src/glum/_glm.py

```diff
@@ -1328,6 +1334,401 @@ def predict(
+            Array with predictions. Estimated if absent.
+        offset : array-like, optional, default=None
+            Array with additive offsets.
+        sample_weight : array-like, shape (n_samples,), optional, default=None
+            Individual weights for each sample.
+        dispersion : float, optional, default=None
+            The dispersion parameter. Estimated if absent.
+        robust : boolean, optional, default=None
+            Whether to compute robust standard errors instead of normal ones.
+            If not specified, the model's ``robust`` attribute is used.
+        clusters : array-like, optional, default=None
+            Array with clusters membership. Clustered standard errors are
+            computed if clusters is not None.
+        expected_information : boolean, optional, default=None
+            Whether to use the expected or observed information matrix.
+            Only relevant when computing robust std-errors.
+            If not specified, the model's ``expected_information`` attribute is used.
+
+        Returns
+        -------
+        WaldTestResult
+            NamedTuple with test statistic, p-value, and degrees of freedom.
+        """
+
+        num_lhs_specs = sum([R is not None, features is not None])
+        if num_lhs_specs != 1:
+            raise ValueError(
+                "Exactly one of R or features must be specified. "
+                f"Received {num_lhs_specs} specifications."
+            )
+
+        if R is not None:
+            return self._wald_test_matrix(
+                R=R,
+                r=r,
+                X=X,
+                y=y,
+                mu=mu,
+                offset=offset,
+                sample_weight=sample_weight,
+                dispersion=dispersion,
+                robust=robust,
+                clusters=clusters,
+                expected_information=expected_information,
+            )
+
+        if features is not None:
+            return self._wald_test_feature_name(
+                features=features,
+                values=r,
+                X=X,
+                y=y,
+                mu=mu,
+                offset=offset,
+                sample_weight=sample_weight,
+                dispersion=dispersion,
+                robust=robust,
+                clusters=clusters,
+                expected_information=expected_information,
+            )
+
+        raise RuntimeError("This should never happen")
+
+    def _wald_test_matrix(
+        self,
+        R: np.ndarray,
+        r: Optional[np.ndarray] = None,
+        X=None,
+        y=None,
+        mu=None,
+        offset=None,
+        sample_weight=None,
+        dispersion=None,
+        robust=None,
+        clusters: np.ndarray = None,
+        expected_information=None,
+    ) -> WaldTestResult:
+        """Compute the Wald test statistic and p-value for a linear hypothesis.
+
+        The hypothesis tested is ``R @ coef_ = r``. The test statistic follows
```

Comment:
Should mention that: "Under the null hyptothesis, the test statistics follows a chi-squared distribution..." (not under the alternative or at least not against all alternatives).